### ISSUE TITLE
[BUG] Game/Vrcompositor crash with Assetto Corsa Competizione VR (UE4, Nvidia)

### ISSUE THREAD

**Issue by jp7677** [ValveSoftware/SteamVR-for-Linux#397]
**Describe the bug**
Running Assetto Corsa Competizione VR via current stable Proton (5.0-9) and a vrclient with this PR applied https://github.com/ValveSoftware/Proton/pull/4163 let the game start, but at some point in the first few minutes SteamVR (vrcompositor) crashes. Very likely the game crashes first and vrcompositor crashing is a follow up. I'm seeing these lines in dmesg:

```
[11897.915514] NVRM: Xid (PCI:0000:02:00): 13, pid=617873408, Graphics Exception: Class 0x0 Subchannel 0x0 Mismatch
[11897.915517] NVRM: Xid (PCI:0000:02:00): 13, pid=617873408, Graphics Exception: ESR 0x4041b0=0x80000
[11897.915519] NVRM: Xid (PCI:0000:02:00): 13, pid=617873408, Graphics Exception: ESR 0x404000=0x80000002
[11897.915828] NVRM: Xid (PCI:0000:02:00): 13, pid=57164, Graphics Exception: ChID 0064, Class 0000c197, Offset 00000f80, Data 00000000
``` 
The game runs perfectly with the same setup without VR.

**To Reproduce**
1. Install a Proton version that includes a Proton vrclient with https://github.com/ValveSoftware/Proton/pull/4163 applied. Here is such a version: https://github.com/ValveSoftware/Proton/issues/1420#issuecomment-705195039 (Here is a link https://github.com/ValveSoftware/Proton#install-proton-locally that describes how to set up up a custom Proton version).
2. Enter the game Assetto Corsa Competizione from the SteamVR lobby, 
3. Enter the actual game (Main Menu -> Single Race -> Start Session)
4. SteamVR crashes during this process, sometimes in the menu, sometimes when entering gameplay, and shows a notification that it wants to restart. Steam console output shows that a crash dump has been uploaded. `dmesg` shows above output.

**Expected behavior**
SteamVR should not crash.

**System Information (please complete the following information):**
- Distribution: Fedora 32.
- SteamVR version: 1.14.16 (the same behavior happens with Beta 1.15.2)
- Steam client version: Oct 7 2020
- Opted into Steam client beta?: No (the same behavior happens with Beta)
- Graphics driver version:  Nvidia 450.80.02 (the same behavior happens with version 455.28)
- Gist for SteamVR System Information: See https://gist.github.com/jp7677/bd7456737d7012bea35e6c9d3926555b
- Proton version: 5.0-9 (including https://github.com/ValveSoftware/Proton/pull/4163)
- DXVK version: 1.7.2 (from https://github.com/doitsujin/dxvk/releases/tag/v1.7.2)

**Additional context**
See also https://github.com/ValveSoftware/Proton/issues/1420#issuecomment-706139550 and further comments.
According to https://github.com/ValveSoftware/Proton/issues/1420#issuecomment-674580748 the same setup works fine when using AMD/Mesa instead of an Nvidia card.
According to https://github.com/ValveSoftware/Proton/issues/1420#issuecomment-706723540 a very similar issue can be observed in Project Cars 2. Both games use Unreal Engine 4.

**Note:** Commenters who are also experiencing this issue are encouraged to include the "System Information" section in their replies.

**Comment by gotzl** [comment]
I get a crash as well (GTX1060 + Index). For me, the sequence is like this
* game window shows up (thanks to mcoffins patch)
* in the garage, it renders a couple of frames (10 - 40), then it freezes and the same Xid messages as OP appear
* Steam logs say `Warning: The game hasn't rendered a frame from us in over 10 seconds`. SteamVR ist still running, headtracking works. ACC log says `[2020.10.11-09.23.11:433][ 18]LogD3D11RHI: Timed out while waiting for GPU to catch up. (0.5 s)`
* after some time, UE gives up and notifies about a crash with a message box (that displays only an 'ok' button). Corresponding ACC log entry seems to be `[2020.10.11-09.25.33:452][ 19]LogWindows: Error: GameThread timed out waiting for RenderThread after 120.00 secs`
* when hitting 'OK', vrcompositor crashes

Is this about the same what you observe? To me, it looks like the vrcompositor crash is more a consequence of the UE crash. The actual cause seems to be the Xid error which seems to result in the GPU not spitting out frames anymore.... 

[Here](https://paste2.org/BND45JnF) the gdb bt of the vrcompositor coredump and the [AC2.log](https://paste2.org/gU4ZC825).

**System Information**
* Distribution: gentoo
* SteamVR version: Beta 1.15.2
* Steam client version: Oct 8 2020
* Opted into Steam client beta?: Yes
* Graphics driver version: Nvidia 450.66
* Proton version: 5.0-9 (including ValveSoftware/Proton#4163)

**Comment by jp7677** [comment]
Thanks for the confirmation, yes this seems to be the same issue. Your conclusion regarding which process crashes first is likely correct.

**Comment by jp7677** [comment]
Above behavior is still the same with Proton 5.13-1 (it still needs the vrclient PR applied manually), Nvidia driver 455.28 and SteamVR 15.4 Beta.

**Comment by jp7677** [comment]
Hi @gotzl Could you please give Proton 5.13-2 RC a try? See https://github.com/ValveSoftware/Proton/issues/4360
I have only tested for a short time, but this seemed to have fixed this issue here for me. Would be cool if you could confirm this.

**Comment by gotzl** [comment]
Thanks for pinging me! It works for me too! Really great :)  I'm on 455.28 ... 
I've also tested only for short time. Will give it some more time later, but the GTX1060 is probably just not enough to get playable FPS ... Still, very good progress!

**Comment by jp7677** [comment]
Thanks for the confirmation and for providing your driver version. I guess then that indeed the Proton RC fixed this. Very cool progress indeed and my GTX is also way to slow, I’m keeping an eye now on the GPU market to see how the new AMD and Nvidia generations unfold :)

Let’s keep this issue open for a short while, though I guess we can close this once Proton 5.13-2 is final.

**Comment by jp7677** [comment]
This issue is resolved with Proton 5.13-2.
